Add instructions for setting maxfiles limit

twangboy opened the pull request:
### What does this PR do?
Adds some instructions for setting the `maxfiles` limit on MacOS so that the zeroMQ tests will pass.

krionbsd commented on pkg/osx/build_env.sh:
```diff
@@ -195,7 +231,7 @@ cd $PKG_DIR
 ./configure --prefix=$PYENV_ROOT
 echo -n -e "\033]0;Build_Env: zeromq: make\007"
-$MAKE -j4
+$MAKE
```
maybe using something like 
```
$MAKE -j `sysctl -n hw.ncpu`
``` 
would make sense

twangboy replied:
OK

twangboy committed:
Updates to build_env script

Add instructions for setting maxfiles limit
Use sysctl to get the number of CPU for the -j make option
